src/genotyper.cc

```diff
@@ -664,6 +977,19 @@ Status genotype_site(const genotyper_config& cfg, MetadataCache& cache, BCFData&
         update_orig_calls_for_loss(cfg, records, bcf_nsamples, dataset_header.get(),
                                    sample_mapping, loss_trackers);
 
+        // Update format fields
+        for (auto& record : records) {
+            vector<int> allele_mapping(record->n_allele, -1);
+            vector<bool> deletion_allele(record->n_allele, false); // which alleles are deletions
+
+            S(find_allele_mapping(site, record.get(), allele_mapping, deletion_allele))
+
+            for (auto& format_helper : format_helpers) {
+                format_helper->add_record_data(dataset, dataset_header.get(), record.get(),
+                                               sample_mapping, allele_mapping);
+            }
+        }
```

Comment:
Just for readability/followability of the code, I'd move this to after the genotypes are computed (around the same place we handle the residuals)